src/main/resources/com/puppycrawl/tools/checkstyle/grammar/java.g

```diff
@@ -543,6 +547,58 @@ annotationExpression
         {#annotationExpression = #(#[EXPR,"EXPR"],#annotationExpression);}
     ;
 
+
+recordDefinition![AST modifiers]
+    :   r:"record" id:id
+        (tp:typeParameters)?
+        rc:recordComponentsList
+        ic:implementsClause
+        rb:recordBodyDeclaration
+        {#recordDefinition = #(#[RECORD_DEF, "RECORD_DEF"],
+                              modifiers, r, id, rc);}
```

Comment:
we not adding tp, ic rb because it is incomplete? I noticed the grammar print had nothing.

Reply:
>we not adding tp, ic rb because it is incomplete?

That is correct, but I still needed to include them here to correctly parse everything.

Reply:
Ok, it may just be me but it sounds like it would be beneficial to add a comment saying updates will come in 8267 to ensure we don't miss anything.

Reply:
I will make it more clear.